dashboard/app/controllers/levels_controller.rb

```diff
@@ -330,6 +330,7 @@ def level_params
     # Reference links should be stored as an array.
     if params[:level][:reference_links].is_a? String
       params[:level][:reference_links] = params[:level][:reference_links].split("\r\n")
+      params[:level][:reference_links].delete_if {|link| link == ""} if params[:level][:reference_links]
```

Comment:
you shouldn't need the trailing `if ...` here. `split` will always return an array.